### PR TITLE
[C++/en] More about references

### PR DESCRIPTION
**Pull request by geoffliu**
Gotta love C++, can't have a simple tutorial and explain basic concepts without getting into the nitty gritty.

### CODE REVIEW

**Comment by levibostian** [comment]
:+1: Thanks!

File: c++.html.markdown [hunk 1]
```diff
@@ -249,6 +250,56 @@ const string& barRef = bar; // Create a const reference to bar.
...
+// Sidetrack: Before we talk more about references, we must introduce a concept
+// called a temporary object. Suppose we have the following code:
+string tempObjectFun() { ... }
+string retVal = tempObjectFun();
+
+// What happens in the second line is actually:
+//   - a string object is returned from `tempObjectFun`
+//   - a new string is constructed with the returned object as arugment to the
+//     constructor
+//   - the returned object is destroyed
```
Comment (vendethiel) [inline c++.html.markdown]: RVO is going to surprise people now :)

Comment (geoffliu) [inline c++.html.markdown]: Oh good call. Hmm I'll think of a different wording then.

Comment (geoffliu) [inline c++.html.markdown]: https://github.com/adambard/learnxinyminutes-docs/commit/85f6ba0b57b9d894c694df66449b1e1c555c625b

@vendethiel How's this? I accidentally pushed directly, didn't make a PR. Feel free to comment on the commit itself.

**Pull request merged**